### ISSUE TITLE
Support eslint-plugin-react-render-types

### ISSUE THREAD

**Issue by yzhe554** [oxc-project/oxc#19198]
jsx doesn't check children's type and nothing is reported by LSP.
In eslint, https://github.com/HorusGoul/eslint-plugin-react-render-types can handle it but it requires typescript-eslint as the parser.

It would be great if oxlint can support this rule or provide a way that we can implement a custom rule to get the same outcome.

**Comment by yzhe554** [comment]
Apologise for raising the issue. I think it's already supported. I got a working version.
Closing the issue.

**Comment by yzhe554** [comment]
The way I made it working is to have a resolver in js which can read the implementation of the component and its jsDoc. This is slow if the project grows too big. It would be great if there is an api from OXLint for this.
Any suggestion? @Boshen @camc314 
Feel free to reopen this issue if you think the request is reasonable.

**Comment by camc314** [comment]
Could you share more of the code you've got, and the use cases?

**Comment by yzhe554** [comment]
@camc314 
The idea is to use jsdoc to config the allowed children and then use js-plugin to lint.
e.g
```
/**
 * @allowedChildren span
 */
export const TestContainer = ({ children }: TestContainerProps) => {
	return <div className="test-container-a">{children}</div>;
};


/**
 * @allowedChildren div
 */
export const TestContainer = ({ children }: TestContainerProps) => {
	return <div className="test-container-b">{children}</div>;


import { TestContainer as TestContainerA } from './variant-a/TestContainer';
import { TestContainer as TestContainerB } from './variant-b/TestContainer';

export const Consumer = () => {
	return (
		<>
			<TestContainerA>
				<span>Content A</span>
			</TestContainerA>
			<TestContainerB>
				<div>Content B</div>
			</TestContainerB>
		</>
	);
};
```
in my custom rule, it is using a custom resolver to find the component and its jsdoc to understand the allowed children then validate. This is because js-plugins doesn't have an api yet from my understanding.

I do agree the best way to implement this might be in tsgo eslint.

Just wondering if js-plugins will expose an api in the future for this use case. Or if you can implement a js-plugin that will be run with --type-aware rules so you can expose more apis.
